core/src/Platforms/iOS/iOSPlatformProxy.cs

```diff
@@ -72,5 +83,60 @@ public string GetDeviceModel()
         {
             return UIDevice.CurrentDevice.Model;
         }
+
+        /// <inheritdoc />
+        public void ValidateRedirectUri(Uri redirectUri, RequestContext requestContext)
+        {
+            if (redirectUri == null)
+            {
+                throw new ArgumentNullException(nameof(redirectUri));
+            }
+
+            if (_isMsal)
+            {
+                if (Constants.DefaultRedirectUri.Equals(redirectUri.AbsoluteUri, StringComparison.OrdinalIgnoreCase))
+                {
+                    // TODO: Need to use CoreExceptionFactory here...?
+                    //throw new MsalException(MsalError.RedirectUriValidationFailed, "Default redirect URI - " + Constants.DefaultRedirectUri +
+                    //                                                               " cannot be used on iOS platform");
+                    throw new InvalidOperationException($"Default redirect URI - {Constants.DefaultRedirectUri} cannot be used on iOS platform");
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public string GetRedirectUriAsString(Uri redirectUri, RequestContext requestContext)
+        {
+            return redirectUri.OriginalString;
+        }
+
+        /// <inheritdoc />
+        public string GetDefaultRedirectUri(string correlationId)
+        {
+            return _isMsal ? string.Format(IosDefaultRedirectUriTemplate, correlationId) : Constants.DefaultRedirectUri;
+        }
+
+        public string GetProductName()
+    {
+        return _isMsal ? "MSAL.Xamarin.iOS" : "PCL.iOS";
+    }
+
+        /// <inheritdoc />
+        public ILegacyCachePersistance CreateLegacyCachePersistence()
+        {
+            return new iOSLegacyCachePersistance();
+        }
+
+        /// <inheritdoc />
+        public ITokenCacheAccessor CreateTokenCacheAccessor()
+        {
+            return new iOSTokenCacheAccessor();
+        }
+
+        /// <inheritdoc />
+        public ICryptographyManager CreateCryptographyManager()
+        {
+            return new iOSCryptographyManager();
+        }
```

Comment:
thinking... should this be cached? We considering the same for some of the other objects. however perhaps this doesn't pay off as those are relatively cheap objects to create.

Reply:
yes, I've changed these to be class initialized properties so they're only constructed once.